### PR TITLE
DEVPROD-2270: Update Ubuntu/Go version, remove Arch Linux.

### PR DESCRIPTION
**Pull request by johndaniels**
This also fixes some underlying bugs with the collectors that were causing the tests to fail.

### CODE REVIEW

File: evergreen.yaml [hunk 1]
```diff
@@ -115,49 +115,37 @@ task_groups:
...
       - lintGroup
       - verify-mod-tidy
 
   - name: ubuntu
-    display_name: Ubuntu 18.04
+    display_name: Ubuntu 22.04
     expansions:
-      GOROOT: /opt/golang/go1.16
+      GOROOT: /opt/golang/go1.20
+      RACE_DETECTOR: true
```
Comment (julianedwards) [inline evergreen.yaml:132]: Do we always want to run the race detector on this? Should we have a variant that runs the same tests without it?

File: collector_buffered.go [hunk 2]
```diff
@@ -27,7 +27,6 @@ func NewBufferedCollector(ctx context.Context, size int, coll Collector) Collect
 		for {
 			select {
 			case <-ctx.Done():
```
Comment (julianedwards) [inline collector_buffered.go:29]: We can remove the select statement logic here by calling `close(c.pipe)` after L47 below and replace the for loop with:
```
for in := c.pipe {
        c.catcher.Add(c.Collector.Add(in))
}
```

**Review by julianedwards — CHANGES_REQUESTED** [review]

File: collector_buffered.go (hunk 2 above)
Comment (johndaniels) [reply collector_buffered.go:29]: I didn't want to do that in this change since I was just trying to do the minimum to get things working. In this case the reading goroutine shouldn't be closing anything so I got rid of the close statement.

File: evergreen.yaml (hunk 1 above)
Comment (johndaniels) [reply evergreen.yaml:132]: Why would we need a variant that does the same tests without it? We need to run the race detector on some variant and that seems like it would just duplicate tests?

Comment (julianedwards) [reply evergreen.yaml:132]: Yeah I thought it was because running the tests with the race detector would be slower (we separate it out for Evergreen but not Cedar). I think this is fine then.

**Review by julianedwards — APPROVED** [review]
